source/jormungandr/jormungandr/olympic_site_params_manager.py

```diff
@@ -150,31 +150,33 @@ def load_olympic_site_params(self, file_path, instance_name):
     def build(self, pt_object_origin, pt_object_destination, api_request, instance):
         # Warning, the order of functions is important
         # Order 1 : get_olympic_site_params
-        # Order 2 : add_criteria
-        # Oredr 3 : add_additional_parameters
+        # Order 2 : build_api_request
         api_request["olympic_site_params"] = self.get_olympic_site_params(
             pt_object_origin, pt_object_destination, api_request, instance
         )
 
-        self.add_criteria(api_request)
-        self.add_additional_parameters(api_request)
+        self.build_api_request(api_request)
 
-    def add_criteria(self, api_request):
+    def build_api_request(self, api_request):
         olympic_site_params = api_request.get("olympic_site_params")
         if not olympic_site_params:
             return
+        # Add keep_olympics_journeys parameter
+        if api_request.get("_keep_olympics_journeys") is None and any(
+            [olympic_site_params.get("departure_scenario"), olympic_site_params.get("arrival_scenario")]
+        ):
```

Comment:
i'm not very sure, but isn't `olympic_site_params` supposed to be `None`, if neither `from` nor `to` is olympics site?

```suggestion
        if api_request.get("_keep_olympics_journeys") is None and olympic_site_params:
```

Reply:
:+1:  Done